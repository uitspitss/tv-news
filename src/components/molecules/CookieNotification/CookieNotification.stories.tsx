import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import base from 'paths.macro';

import {
  CookieNotification,
  CookieNotificationProps,
} from './CookieNotification';

const path = base.replace(
  /^.+\/(atoms|molecules|organisms|templates|pages)\/(.+)\//,
  '$1/$2',
);

export default {
  title: `${path}`,
  component: CookieNotification,
  args: {
    debug: true,
  },
} as Meta;

const Template: Story<CookieNotificationProps> = (args) => (
  <CookieNotification {...args} />
);

export const Default = Template.bind({});
