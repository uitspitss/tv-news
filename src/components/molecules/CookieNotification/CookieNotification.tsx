import React, { memo, VFC, useCallback } from 'react';
import tw, { styled } from 'twin.macro';
import CookieConsent from 'react-cookie-consent';

export type CookieNotificationProps = {
  /**
   * class for twin.macro
   */
  className?: string;
  /**
   * whether debug
   */
  debug?: boolean;
};

type Props = {
  handleDecline: () => void;
} & CookieNotificationProps;

const Component: VFC<Props> = (props) => {
  const { className, debug = false } = props;

  return (
    <CookieConsent
      disableStyles
      flipButtons
      location="bottom"
      containerClasses={className}
      buttonClasses="acceptButton"
      buttonText="理解しました"
      // enableDeclineButton
      // declineButtonClasses="declineButton"
      // onDecline={handleDecline}
      debug={debug}
    >
      この Web サイトではユーザー体験向上のため cookie を使用しております。
    </CookieConsent>
  );
};

const StyledComponent = styled(Component)`
  ${tw`fixed z-10 flex flex-wrap w-full justify-between items-center bg-primary-600 text-white text-sm font-bold px-4 py-3`}

  & > div {
    & > .acceptButton {
      ${tw`bg-info-700 hover:bg-info-800 text-white font-bold py-2 px-4 mr-2 border-none cursor-pointer`}
    }
    & > .declineButton {
      ${tw`bg-transparent no-underline hover:underline text-white font-bold py-2 px-4 border-none cursor-pointer`}
    }
  }
`;

/**
 * cookie の利用確認用 notification
 */
export const CookieNotification = memo((props: CookieNotificationProps) => {
  const handleDecline = useCallback(() => {
    // Cookies.remove('_ga');
    // Cookies.remove(`_ga_${GA_TRACKING_ID}`);
  }, []);

  return <StyledComponent {...props} {...{ handleDecline }} />;
});
