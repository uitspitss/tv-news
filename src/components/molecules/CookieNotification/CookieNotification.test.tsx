import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';

import { CookieNotification } from './CookieNotification';

describe('molecules/CookieNotification', () => {
  test('snapshot test', () => {
    const { container } = render(<CookieNotification />);
    expect(container).toMatchSnapshot();
  });

  describe('Behavior', () => {
    beforeEach(() => {
      jest.clearAllMocks();
      render(<CookieNotification />);
    });

    test('initial view', () => {
      expect(
        screen.getByText(
          'This website uses cookies to enhance the user experience.',
        ),
      ).toBeInTheDocument();
      expect(screen.getByText('I understand')).toBeInTheDocument();
    });
    test('click, then invisible', () => {
      fireEvent.click(screen.getByText('I understand'));

      expect(
        screen.queryByText(
          'This website uses cookies to enhance the user experience.',
        ),
      ).not.toBeInTheDocument();
      expect(screen.queryByText('I understand')).not.toBeInTheDocument();
    });
  });
});
