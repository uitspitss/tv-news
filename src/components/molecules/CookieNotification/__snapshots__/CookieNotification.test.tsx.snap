// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/CookieNotification snapshot test 1`] = `
.emotion-0 {
  position: fixed;
  z-index: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  --tw-bg-opacity: 1;
  background-color: rgba(75, 85, 99, var(--tw-bg-opacity));
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: 700;
  padding-left: 1rem;
  padding-right: 1rem;
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
}

.emotion-0>div>.acceptButton {
  --tw-bg-opacity: 1;
  background-color: rgba(29, 78, 216, var(--tw-bg-opacity));
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  font-weight: 700;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  margin-right: 0.5rem;
  border-style: none;
  cursor: pointer;
}

.emotion-0>div>.acceptButton:hover {
  --tw-bg-opacity: 1;
  background-color: rgba(30, 64, 175, var(--tw-bg-opacity));
}

.emotion-0>div>.declineButton {
  background-color: transparent;
  -webkit-text-decoration: none;
  text-decoration: none;
  --tw-text-opacity: 1;
  color: rgba(255, 255, 255, var(--tw-text-opacity));
  font-weight: 700;
  padding-top: 0.5rem;
  padding-bottom: 0.5rem;
  padding-left: 1rem;
  padding-right: 1rem;
  border-style: none;
  cursor: pointer;
}

.emotion-0>div>.declineButton:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <div
    class="emotion-0 emotion-1"
    style="bottom: 0px;"
  >
    <div
      class=""
    >
      この Web サイトではユーザー体験向上のため cookie を使用しております。
    </div>
    <div
      class=""
    >
      <button
        aria-label="Accept cookies"
        class="acceptButton"
        id="rcc-confirm-button"
      >
        理解しました
      </button>
    </div>
  </div>
</div>
`;
